app/routes/items/new.js

```diff
@@ -69,78 +54,98 @@ export default AuthorizeRoute.extend({
     });
   },
 
-  async setupController(controller, model) {
+  setupController(controller, model) {
     this._super(controller, model);
+    const store = this.get("store");
+    this.initializeController();
+
+    if (this.get("newItemRequest")) {
+      this.initializeDimensions();
+      this.manageSubformDetails();
+      this.setUpPackageImage();
+      window.localStorage.setItem("isSelectLocationPreviousRoute", false);
+    }
+  },
 
-    controller.set("inventoryNumber", this.get("inventoryNumber"));
-    controller.set("displayInventoryOptions", false);
-    controller.set("autoGenerateInventory", true);
-    controller.set("inputInventory", false);
+  initializeController() {
+    const controller = this.controller;
+    if (!controller.get("inventoryNumber")) {
+      controller.send("autoGenerateInventoryNumber");
+    }
     controller.set("invalidLocation", false);
     controller.set("invalidScanResult", false);
     controller.set("labels", 1);
+  },
 
-    if (this.get("newItemRequest")) {
-      this.set("newItemRequest", false);
-      controller.set("quantity", 1);
-      if (
-        window.localStorage.getItem("isSelectLocationPreviousRoute") === "false"
-      ) {
-        controller.set("caseNumber", "");
-        controller.set("length", null);
-        controller.set("width", null);
-        controller.set("height", null);
-        controller.set("selectedGrade", {
-          name: "B",
-          id: "B"
-        });
-        controller.set("imageKeys", "");
-      }
-      let codeId = controller.get("codeId");
-      if (codeId) {
-        let selected = this.get("store").peekRecord("code", codeId);
-        if (selected) {
-          let selectedSubform = selected.get("subform");
-          if (this.isSubformAllowed(selectedSubform)) {
-            controller.set("showAdditionalFields", true);
-            let details = await this.store.query(selectedSubform, {
-              distinct: "brand"
+  async manageSubformDetails() {
+    const controller = this.controller;
```

Comment:
Tip: to avoid endless nesting of IFs, you can use negatives and returns, which usually leads to simpler and more readable code

From
```javascript
if (A) {
  if (B) {
    if (C) {
        // logic
    }
  }
}
```

to 
```javascript
if (!A) return or throw;
if (!B) return or throw;
if (!C) return or throw;

// logic
```